App/dojo.py

```diff
@@ -369,8 +386,149 @@ def load_people(self, filename):
+        # connect to db
+        Base.metadata.bind = engine
+        Session = sessionmaker()
+        session = Session()
+
+        # function to select all rooms in the Rooms table
+        items = select([Rooms])
+        res = session.execute(items)
+        db_room_list =[item.room.name for item in res]
+
+        for room in self.all_rooms:
+            if room.name not in db_room_list:
+                new_room = Rooms(name=room.name,
+                                type_room = room.room_type,room_capacity = room.capacity)
+                session.add(new_room)
+                session.commit()
+
+        people = select([Persons])
+        response = session.execute(people)
+        dbpersons_list = [item.person.name for item in response]
+        # go through all people in all people list
+        for person in self.all_people:
+            if person.name not in dbpersons_list:
+                # get the neccessary office names from objects
+                if person.office is None:
+                    # If fellow or staff have no offices allocated
+                    people.office = 'has no office'
+                else:
+                  office_allocated = person.office.name
+                #    If fellows have no livingspaces allocated
+                if person.livingspace is None:
+                    livingspace_allocated = "no room"
+                else:
+                  livingspace_allocated= person.livingspace.name
+                accomodation = person.wants_accomodation
+                new_p = Persons(
+                    name=person.name,
+                    category=person.person_type,
+                    wants_accomodation= accomodation,
+                    office_allocated= office_allocated,
+                    living_space_allocated= livingspace_allocated)
+                # add new person to db
+                session.add(new_p)
+                session.commit()
+                message = "Data added to {} database successfully"
+                click.secho(message, fg='green', bold=True)   
 
-    def save_state(self,db_name):
-        pass
     def load_state(self, db_name):
-        pass
\ No newline at end of file
+        '''
+         This function loads data from a database into the application.
+        '''
+        if not os.path.isfile(db_name):
+            print("Database does not exist")
+        else:    
+            # create engine
+            engine = create_engine('sqlite:///' + db_name)
+            # Bind engine to Base Metadata
+            Base.metadata.bind = engine
+            # creates session
+            Session = sessionmaker(bind=engine)
+            session = Session()
+            # Rooms --->
+            # gets all the rooms in Rooms
+            items = select([Rooms])
+            # create an instance of the Rooms class where data is loaded
+            res = session.execute(items)
+            # queries all the result to enable tabulation of data
```

Comment:
there's a more efficient way to query the database
http://docs.sqlalchemy.org/en/latest/orm/tutorial.html#querying